viewing_party/party.py

```diff
@@ -0,0 +1,145 @@
+                rating = user_data["watched"][i]["rating"]
+                sum += rating
+                i += 1
+                average_rating = (sum / len(user_data["watched"]))
+            return average_rating
+    if len(user_data["watched"]) == 0:
+        average_rating = 0.0
+        return average_rating
+
+
+
+def get_most_watched_genre(user_data):
+    count_genre = []
+    i = 0
+
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for i in range(len(user_data["watched"])):
+            genre = user_data["watched"][i]["genre"]
+            count_genre.append(genre)
+
+        most_watched_genre = max(count_genre, key = count_genre.count)
+        return most_watched_genre
+
+
+
+#WAVE 3
+def get_unique_watched(user_data):
+    unique_user_watched = []
+    friends_watched = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)
+
+    for movie in user_data["watched"]:
+        if movie not in friends_watched:
+            unique_user_watched.append(movie)
+    return unique_user_watched
+
+
+
+def get_friends_unique_watched(user_data):
+    unique_friends_watched = []
+    user_watched = []
+
+    for movie in user_data["watched"]:
+            user_watched.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]: 
+            if movie["title"] not in user_watched:
+                if movie not in unique_friends_watched: 
+                    unique_friends_watched.append(movie)
+    return unique_friends_watched
+
+
+
+#WAVE 4
+def get_available_recs(user_data):
+    friends_unique_movies = get_friends_unique_watched(user_data)
+    user_recommendations = []
+
+
+    for movie in friends_unique_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            if movie not in user_recommendations:
+                user_recommendations.append(movie)
+    return user_recommendations
+
+
+
+#WAVE 5
+def get_new_rec_by_genre(user_data):
+    user_most_watched_genre = get_most_watched_genre(user_data)
+    user_recs_based_on_genre = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"]:
+                if movie["genre"] == user_most_watched_genre:
+                    if movie not in user_recs_based_on_genre:
+                        user_recs_based_on_genre.append(movie)
+    return user_recs_based_on_genre
+
+
+
+def get_rec_from_favorites(user_data):
+    friends_watched = []
+    user_recs_from_favs = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)
+
+    for movie in user_data["favorites"]:
+        if movie not in friends_watched:
+            if movie not in user_recs_from_favs:
+                user_recs_from_favs.append(movie)
+    return user_recs_from_favs
```

Comment:
💡 Quick-Tip: To make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.